packages/components/addon/components/hds/tabs/index.js

```diff
@@ -0,0 +1,90 @@
+import Component from '@glimmer/component';
+import { A } from '@ember/array';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class HdsTabsIndexComponent extends Component {
+  // UNCOMMENT THIS IF YOU NEED A CONSTRUCTOR
+  // constructor() {
+  //   super(...arguments);
+  //   // ADD YOUR ASSERTIONS HERE
+  // }
+
+  @tracked tabNodes = A([]);
+  @tracked tabIds = A([]);
```

Comment:
The element id is used to find its nodeIndex in the tab/panel id arrays so that it can determine if it's the selected element which sets the selected tab class name and aria attributes, etc.

It may be possible to get rid of the tabIds & panelIds arrays and instead use the element.id in a search on the node arrays but it seems in order to do that, I'd need to use the plain JavaScript array vs. the Ember array. Using the plain array was breaking other things when I tried using it like in here though: https://github.com/hashicorp/design-system/blob/79047c1c088e1cbc40e85a89a92a108c75346ec1/packages/components/addon/components/hds/form/field/index.js#L13

I'll continue to try to look at this though. (I haven't looked through your branch in detail yet to see if you did this.)

Reply:
I pushed up some refactors to my initial code based on some of your other feedback and your example branch. The way I am still currently doing things though I can't really sea a way to get rid of the id arrays.